jupyter-lib/kotlin-jupyter-api/src/main/kotlin/org/jetbrains/kotlin/jupyter/api/libraryConnection.kt

```diff
@@ -0,0 +1,47 @@
+package org.jetbrains.kotlin.jupyter.api
+
+typealias TypeName = String
+typealias Code = String
+
+data class TypeHandler(val className: TypeName, val code: Code)
+
+open class LibraryDefinition(
```

Comment:
Why open class and not an interface with default methods? Also, this definition does not to create definitions directly from the code. We would like to create a Library definition as a class without and String code injections and use it directly. Maybe it is better to create a definition interface that uses functions instead of `Code`  and an abstract class that uses code blocks to implement those functions.

Reply:
Changed to something similar. Now definition should return the lists of `Execution` objects, they may be either simple functions or code snippets (instances of `CodeExecution` in the latter case)